src/components/PopupWithForm.js

```diff
@@ -21,24 +23,38 @@ export default class PopupWithForm extends Popup {
       inputValues[input.name] = input.value;
     });
     return inputValues;
-
-    //which collects data from all the input fields and returns it as an object.
-    //This data should then be passed to the submission handler as an argument.
+    // Collects data from all the input fields and returns it as an object.
+    // This data should be passed to the submission handler as an argument.
   }
 
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
+
+      // Change the button text and disable it while processing
+      this._submitButton.textContent = "Saving...";
+      this._submitButton.disabled = true;
       const formData = this._getInputValues();
-      this._handleFormSubmit(formData);
-      this._popupForm.reset();
-      this.close();
+      this._handleFormSubmit(formData)
+        .then(() => {
+          // After submission completes, reset button state
+          this._submitButton.textContent = "Submit";
```

Comment:
`          this._submitButton.textContent = "Submit";
`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`. You shouldn't duplicate it in `then` and `catch`